include/dataformat.hxx

```diff
@@ -118,12 +124,37 @@ class DataFormat {
   ///
   /// Inputs
   /// ------
-  /// 
+  ///
   /// @param[in] varname     Variable name. The variable must already exist
   /// @param[in] attrname    Attribute name
   /// @param[in] value       A string attribute to attach to the variable
   virtual void setAttribute(const string &UNUSED(varname), const string &UNUSED(attrname),
                             int UNUSED(value)) {}
+  /// Gets a string attribute
+  ///
+  /// Inputs
+  /// ------
+  ///
+  /// @param[in] varname     Variable name. The variable must already exist
+  /// @param[in] attrname    Attribute name
+  ///
+  /// Returns
+  /// -------
+  /// text                   A string attribute of the variable
+  virtual bool getAttribute(const string &UNUSED(varname), const string &UNUSED(attrname), std::string &UNUSED(text)) {}
```

Comment:
This can be declared pure (` = 0`), and the `UNUSED` macros removed, likewise for the other overload. This currently raises a warning about no return value from a non-void function.

`setAttribute` above could also be declared pure.